sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java

```diff
@@ -325,6 +355,13 @@ public boolean matches(String filename) {
       return toBuilder().setCompressionType(compressionType).build();
     }
 
+    /**
+     * Sets the XML file charset.
+     */
+    public Read<T> withCharset(String charset) {
```

Comment:
`Charset` is not `Serializable`, that's why I preferred to use a simple `String`.